convert_representation.py

```diff
@@ -8,23 +12,158 @@ def convert_parenthesized_to_tree(parenthesized: str) -> Graph:
     
     Args:
         parenthesized (str): Reprezentacja nawiasowa struktury RNA.
-    ValueError: Kiedy nie da sie przekonwertowac, bo struktura jest zla
-    
+    Raises:
+        ValueError: Kiedy nie da sie przekonwertowac, bo struktura nawiasów jest zla
     Returns:
-        Graph: Graf reprezentujący strukturę RNA.
+        Graph: Graf reprezentujący strukturę RNA. 
+        Struktura tego grafu jest opisana w README.md
     """
-    # TODO: treść funkcji
-    pass
+    g = ig.Graph(directed=True)
+    v = g.add_vertex(is_root=True, unpaired_count_0=0, unpaired_count_1=0, unpaired_count_2=0)
+    opened_parentheses_count = 0
+
+    for c in parenthesized:
+        if c == ".":
+            v[f"unpaired_count_{v.outdegree()}"] = (v.attributes().get(f"unpaired_count_{v.outdegree()}") or 0) + 1
+        if c == "(":
+            opened_parentheses_count += 1
+            v_prev = v
+            v = g.add_vertex(is_root=False, unpaired_count_0=0, unpaired_count_1=0)
+            g.add_edge(v_prev, v)
+        if c == ")":
+            opened_parentheses_count -= 1
+            if (len(g.predecessors(v)) == 0):
+                raise ValueError("Unopened parenthesis")
+            v = g.vs[g.predecessors(v)[0]]
```

Comment:
Jak szukanie przodków wpłynie na złożoność czasową? Czy chcemy się tym przejmować?

Reply:
Bardzo słuszna uwaga.

2) Tutaj przy konwersji reprezentacji myślę że nie warto się tym zajmować, bo to malutki ułamek pracy którą wykonuje nasz program.

1) Myślę że nie zmieni, bo wierzę że twórcy biblioteki mają to zoptymalizowane.

To pewnie będzie super  ważne w przypadku algorytmu oceniającego projektowalność. Po wstępnym napisaniu algorytmu oceny projektowalności się tym zajmę i zobaczę czy faktycznie spowalnia program. (Z mojego doświadczenia najlepiej robić optymalizację już po napisaniu algorytmu, ponieważ wtedy można profilować program i zobaczyć na żywo gdzie schodzi najwięcej cykli procesora.)

Reply:
@WildJellybear  Po odczytaniu kliknij proszę na "resolve" pod tą konwersacją żeby ta rozmowa zniknęła i żebym widział że przeczytałeś. (chyba że chcesz coś dodać, to śmiało)